src/Symfony/Component/Config/Definition/BaseNode.php

```diff
@@ -340,7 +340,7 @@ final public function normalize(mixed $value): mixed
 
         // replace value with their equivalent
         foreach ($this->equivalentValues as $data) {
-            if ($data[0] === $value) {
+            if ($data[0] === ($value instanceof \Stringable ? (string) $value : $value)) {
```

Comment:
TODO: this will change the behavior of all nodes so the cast should be moved to `ScalarNode`

Reply:
shouldn't this logic be moved to preNormalize?